### PR TITLE
[feat] merge relative date refiner

### PR DESCRIPTION
**Pull request by liamcain**
## Summary

Adds support for more complex composite dates.

 - 2 weeks before 2020-02-13
 - 2 days after next Friday

### CODE REVIEW

**Comment by wanasit** [comment]
Hey @liamcain. Thank you for your work! 
Could you please check my comment and tweak your change a bit?

**Comment by liamcain** [comment]
Hi @wanasit, is there a review you forgot to submit? I don't see the comment you're referring to. Also just noticed that I had `prettier` changes unstaged, so I just pushed a fix for that.

**Comment by wanasit** [comment]
> I don't see the comment you're referring to. Also just noticed that I had `prettier` changes unstaged, so I just pushed a fix for that.

Ops. Sorry, I'm actually not very used to CR through Github ^^;

File: src/locales/en/refiners/ENMergeDateRelativeRefiner.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ * - 2 days after next Friday
+ */
+export default class ENMergeDateRelativeRefiner extends MergingRefiner {
+    patternBetween(): RegExp {
+        return /^\s+$/i;
+    }
+
+    shouldMergeResults(textBetween: string, currentResult: ParsingResult, nextResult: ParsingResult): boolean {
+        return (
+            textBetween.match(this.patternBetween()) != null &&
```
[the remark this answers is not in the document]
Comment (liamcain) [reply src/locales/en/refiners/ENMergeDateRelativeRefiner.ts]: Great point, this was confusing for me to read just a few days after having written it. Improved the code style here and added some more context to make it easier follow.

File: test/en/en_compound_dates.test.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
+import * as chrono from "../../src";
+import { testSingleCase } from "../test_util";
+
+test("Test - Single Expression", function () {
```
[the remark this answers is not in the document]
Comment (liamcain) [reply test/en/en_compound_dates.test.ts]: Updated

File: src/locales/en/refiners/ENMergeDateRelativeRefiner.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import { PATTERN as AGO_PATTERN } from "../parsers/ENTimeUnitAgoFormatParser";
+import { PATTERN as LATER_PATTERN } from "../parsers/ENTimeUnitLaterFormatParser";
+import { reverseTimeUnits } from "../../../utils/timeunits";
+
+/**
+ * Merges an absolute date with a relative date.
+ * - 2 weeks before 2020-02-13
+ * - 2 days after next Friday
+ */
+export default class ENMergeDateRelativeRefiner extends MergingRefiner {
```
[the remark this answers is not in the document]
Comment (liamcain) [reply src/locales/en/refiners/ENMergeDateRelativeRefiner.ts]: Changed to your suggest. I was also thinking `ENMergeImpliedReferenceDateRefiner` might be more accurate, lmk what you think.

**Pull request merged**